### ISSUE TITLE
Usage examples that do not require metadata

### ISSUE THREAD

**Issue by romanholidaypancakes** [stephenberry/glaze#624]
I noticed that it says no metadata is required

![image](https://github.com/stephenberry/glaze/assets/51652878/fd850111-37a5-433d-815b-c6270795791c)
then I used the following code

```C++
#include "glaze/binary/read.hpp"
#include "glaze/binary/write.hpp"

struct my_struct {
    int                        i     = 287;
    double                     d     = 3.14;
    std::string                hello = "Hello World";
    std::array<uint64_t, 3>    arr   = {1, 2, 3};
    std::map<std::string, int> map{{"one", 1}, {"two", 2}};
};

int main() {
    my_struct              m;
    std::vector<std::byte> b;
    glz::write_binary(m, b);

    my_struct m2;
    auto      xxvf = glz::read_binary(m2, b);
}

```

The compiler gave the error
```
glaze\binary\write.hpp(101,30): error C2338: static_assert failed: 'Glaze metadata is probably needed for your type'
```

I am using binary serialization/deserialization, which in my understanding doesn't need metadata (of field names) because I don't need to serialize/deserialize to json.

**Comment by stephenberry** [comment]
Make sure you are on the main branch, but even then I'll have to add more tests for binary reflection support. I was planning to make a new release today that supports reflection for binary, so it is in progress.

**Comment by stephenberry** [comment]
I'll also note to be wary reading into a std::string_view. You must keep your input buffer alive for the string_view memory to be valid.
Using a std::string is much safer. But, if you are only writing then there are no dangers.

**Comment by stephenberry** [comment]
I just released version 1.9.6, which includes reflection support for binary
